src/api/test-controller/index.js

```diff
@@ -379,6 +382,19 @@ export default class TestController {
     _expect$ (actual) {
         const callsite = getCallsiteForMethod('expect');
 
+        if (this.testRun.observedCallsites) {
```

Comment:
I think we can extract this code to a method (e.g. `checkExcessiveCallsites`) and implement a shortcut (`if (!this.testRun.observedCallsites) return` in it.